packages/chatbot-server-mongodb-public/src/processors/generateResponseWithTools.eval.ts

```diff
@@ -0,0 +1,288 @@
+import "dotenv/config";
+import { Eval, EvalCase, EvalScorer } from "mongodb-rag-core/braintrust";
+import { createAzure } from "mongodb-rag-core/aiSdk";
+import {
+  assertEnvVars,
+  CORE_OPENAI_ENV_VARS,
+  defaultConversationConstants,
+  SomeMessage,
+  FindContentFunc,
+  FindContentFuncArgs,
+  MongoDbPageStore,
+  PersistedPage,
+} from "mongodb-rag-core";
+import {
+  GenerateResponseParams,
+  GenerateResponseReturnValue,
+} from "mongodb-chatbot-server";
+import { systemPrompt } from "../systemPrompt";
+import { makeGenerateResponseWithTools } from "./generateResponseWithTools";
+import { makeMongoDbReferences } from "./makeMongoDbReferences";
+import { makeFetchPageTool } from "../tools/fetchPage";
+import { makeSearchTool } from "../tools/search";
+import { ObjectId } from "mongodb-rag-core/mongodb";
+import { getGenerateWithToolsEvalCasesFromYamlFile } from "../eval/getGenerateWithToolEvalCasesFromYaml";
+
+const { OPENAI_API_KEY, OPENAI_CHAT_COMPLETION_DEPLOYMENT } = assertEnvVars({
+  ...CORE_OPENAI_ENV_VARS,
+  OPENAI_CHAT_COMPLETION_DEPLOYMENT: "",
+});
+
+// Store the desired content for each test in the EvalCase metadata
+type IntermediateToolResponse = {
+  loadPageReturnContent?: string;
+  findContentReturnContent?: string;
+};
+
+type GenerateResponseInput = Partial<GenerateResponseParams> &
+  Pick<GenerateResponseParams, "latestMessageText">;
+
+type GenerateResponseExpectedMessage = {
+  role: SomeMessage["role"] | "assistant-tool";
+  toolCallName?: string;
+  toolCallArgs?: Record<string, string>;
+};
+
+type GenerateResponseExpected = {
+  messages: GenerateResponseExpectedMessage[];
+};
+
+// Load eval cases from a YAML file.
+const evalCases: EvalCase<
+  GenerateResponseInput,
+  GenerateResponseExpected,
+  IntermediateToolResponse
+>[] = getGenerateWithToolsEvalCasesFromYamlFile(
+  require.resolve("../../evalCases/generate_response_with_tools.yml")
```

Comment:
this is a bit of an odd pattern for loading the eval cases. was there a specific reason you chose?

generally prefer using `path` module (`path.resolve(...)`). see other examples in repo for reference, and pls update this to use as well if there wasn't a clear reason for this one.

also as a general practice when loading stuff from file, prefer not to use `/` b/c if using this code on windows, i believe it wouldnt work b/c windows uses `\`. instead do `path.resolve("..", "..", "evalCases" ...)`